### DISCUSSION TITLE
Problem for setting up a tunnel over ipsec strongswan with openwrt

### DISCUSSION THREAD

**Discussion by RusselWestbrook6** [strongswan/strongswan#1265] [Q&A]
Refer to the scene in the issue
https://wiki.strongswan.org/issues/3001

I have to use libipsec in my project.
I relly want to know
1. how to use Libipsec to establish ipsec connection in this scenario.
2. Why does this happen

**Comment by RusselWestbrook6** [comment]
@tobiasbrunner

**Comment by RusselWestbrook6** [comment]
Linux strongSwan U5.7.2/K4.19.26-00788-gbd95801334a2 
Does this kernel version matter？

**Comment by RusselWestbrook6** [comment]
if (hydra->kernel_interface->get_address_by_ts(hydra->kernel_interface, src_ts, &src_ip, &is_virtual) != SUCCESS)

what does this mean

**Comment by tobiasbrunner** [comment]
I have no idea what you are asking, sorry.

**Reply by RusselWestbrook6** [reply]
if (hydra->kernel_interface->get_address_by_ts(hydra->kernel_interface, src_ts, &src_ip, &is_virtual) != SUCCESS)
in strongswan/src/libcharon/plugins/kernel_libipsec: install_route

May I know the reason for this judgment？

**Reply by tobiasbrunner** [reply]
I don't understand, sorry. Also, you are using very old code, libhydra has been removed with 5.4.0, which was released over 6 years ago.

**Comment by Thermi** [comment]
I think it's an issue with the route installation expecting a virtual IP or something, not quite sure. I had a similar issue in the past.

**Comment by RusselWestbrook6** [comment]
thanks to both of you very much
